package/MDAnalysis/coordinates/memory.py

```diff
@@ -0,0 +1,255 @@
+        """
+        Overrides the positions property in base.Timestep to
+        use avoid duplication of the array.
+        """
+
+        @property
+        def positions(self):
+            return base.Timestep.positions.fget(self)
+
+        @positions.setter
+        def positions(self, new):
+            self.has_positions = True
+            # Use reference to original rather than a copy
+            self._pos = new
+
+    _Timestep = MemoryTimestep
+
+    def __init__(self, coordinate_array, format='afc',
+                 dimensions = None, dt=1, **kwargs):
+        """
+
+        Parameters
+        ---------
+        coordinate_array : :class:`~numpy.ndarray object
+            The underlying array of coordinates
+        format : str, optional
+            the order/shape of the return data array, corresponding
+            to (a)tom, (f)rame, (c)oordinates all six combinations
+            of 'a', 'f', 'c' are allowed ie "fac" - return array
+            where the shape is (frame, number of atoms,
+            coordinates)
+        dimensions: (*A*, *B*, *C*, *alpha*, *beta*, *gamma*), optional
+            unitcell dimensions (*A*, *B*, *C*, *alpha*, *beta*, *gamma*)
+
+            lengths *A*, *B*, *C* are in the MDAnalysis length unit (Å), and
+            angles are in degrees.
+        dt: float, optional
+            The time difference between frames (ps).  If :attr:`time`
+            is set, then `dt` will be ignored.
+        """
+
+        super(MemoryReader, self).__init__()        
+
+        self.set_array(np.asarray(coordinate_array), format)
+        self.n_frames = self.coordinate_array.shape[self.format.find('f')]
+        self.n_atoms = self.coordinate_array.shape[self.format.find('a')]
+
+        provided_n_atoms = kwargs.pop("n_atoms", None)
+        if (provided_n_atoms is not None and
+            provided_n_atoms != self.n_atoms):
+                raise ValueError("The provided value for n_atoms does not match"
+                                 "the shape of the coordinate array")
+
+        self.ts = self._Timestep(self.n_atoms, **kwargs)
+        self.ts.dt = dt
+        if dimensions is not None:
+            self.ts.dimensions = dimensions
+        self.ts.frame = -1
+        self.ts.time = -1
+        self._read_next_timestep()
+
+    def set_array(self, coordinate_array, format='afc'):
+        """
+        Set underlying array in desired column format.
+
+        Parameters
+        ---------
+        coordinate_array : :class:`~numpy.ndarray object
+            The underlying array of coordinates
+        format
+            The order/shape of the return data array, corresponding
+            to (a)tom, (f)rame, (c)oordinates all six combinations
+            of 'a', 'f', 'c' are allowed ie "fac" - return array
+            where the shape is (frame, number of atoms,
+            coordinates)
+        """
+        self.coordinate_array = coordinate_array
+        self.format = format
```

Comment:
Well spotted. Although this never leads to problems in practice (the class variable is only read on uninstantiated classes, and the object variables only on instances), it is extremely confusing and ugly. I've renamed `self.format` to `self.stored_format`. Fixed in 2637a20.